packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -241,10 +260,16 @@ class ScrollDragController implements Drag {
   @override
   void update(DragUpdateDetails details) {
     assert(details.primaryDelta != null);
+    final bool lastEventWasStationary = _lastDetails is DragUpdateDetails
+        && _lastDetails.primaryDelta == 0.0;
     _lastDetails = details;
     double offset = details.primaryDelta;
-    if (offset == 0.0)
+    if (offset == 0.0) {
+      // No axis motion over 2 consecutive frames breaks previous momentum.
```

Comment:
Stopwatch can't be mocked so that will be difficult to test. :-)
I recommend depending on the timestamp on the event, that's the number that actually matters for the input data.